converter/biostudies.py

```diff
@@ -0,0 +1,176 @@
+                "value": ['name', _parse_name, 0]
+            },
+            {
+                "name": ['', default_to, "Middle Initials"],
+                "value": ['name', _parse_name, 1]
+            },
+            {
+                "name": ['', default_to, "Last Name"],
+                "value": ['name', _parse_name, 2]
+            },
+            {
+                "name": ['', default_to, "Email"],
+                "value": ['email']
+            },
+            {
+                "name": ['', default_to, "Phone"],
+                "value": ['phone']
+            },
+            {
+                "name": ['', default_to, "Affiliation"],
+                "value": ['institution']
+            },
+            {
+                "name": ['', default_to, "Address"],
+                "value": ['address']
+            },
+            {
+                "name": ['', default_to, "Orcid ID"],
+                "value": ['orcid_id']
+            },
+    ],
+        True
+    ]
+}
+
+PROJECT_SPEC_ORGANIZATIONS = {
+    "type": ['', default_to, "Organization"],
+    "$on": 'funders',
+    "attributes": ['$array', [
+            {
+                "name": ['', default_to, "Grant ID"],
+                "value": ['grant_id']
+            },
+            {
+                "name": ['', default_to, "Grant Title"],
+                "value": ['grant_title']
+            },
+            {
+                "name": ['', default_to, "Organization"],
+                "value": ['organization']
+            },
+    ],
+        True
+    ]
+}
+
+
+class BioStudiesConverter:
+
+    def __init__(self):
+        self.attributes = None
+        self.contributors = None
+        self.funders = None
+        self.publications = None
+
+    def convert(self, hca_project: dict, additional_attributes: dict = None) -> dict:
+        converted_project = JsonMapper(hca_project).map({
+            'attributes': ['$array', PROJECT_SPEC_BASE, True],
+            'section': PROJECT_SPEC_SECTION
+            })
+
+        project_content = hca_project['content'] if 'content' in hca_project else None
+        if project_content:
+            self.__add_subsections_to_project(converted_project, project_content)
+
+        return converted_project
+
+    @staticmethod
+    def __add_subsections_to_project(converted_project, project_content):
+        contributors = project_content.get('contributors')
+        funders = project_content.get('funders')
+        publications = project_content.get('publications')
+        if contributors or funders or publications:
+            converted_project['section']['subsections'] = []
+
+        converted_publications = JsonMapper(project_content).map(PROJECT_SPEC_PUBLICATIONS) if publications else []
+        converted_authors = JsonMapper(project_content).map(PROJECT_SPEC_AUTHORS) if contributors else []
+        converted_funders = JsonMapper(project_content).map(PROJECT_SPEC_ORGANIZATIONS) if funders else []
```

Comment:
I like this fragmentation of the spec files to account for different sections of the BioStudies base spec.